src/aleph/http/server.clj

```diff
@@ -196,42 +199,73 @@
                       {:status 204}
 
                       :else
-                      (invalid-value-response req rsp))))))))))))
-
-(defn exception-handler [ctx ex]
-  (cond
-    ;; do not need to log an entire stack trace when SSL handshake failed
-    (http/ssl-handshake-error? ex)
-    (log/warn "SSL handshake failure:"
-              (.getMessage ^Throwable (.getCause ^Throwable ex)))
-
-    (not (instance? IOException ex))
-    (log/warn ex "error in HTTP server")))
-
-(defn invalid-request? [^HttpRequest req]
-  (-> req .decoderResult .isFailure))
-
-(defn reject-invalid-request [ctx ^HttpRequest req]
-  (d/chain
-    (netty/write-and-flush ctx
-      (DefaultFullHttpResponse.
-        HttpVersion/HTTP_1_1
-        HttpResponseStatus/REQUEST_URI_TOO_LONG
-        (-> req .decoderResult .cause .getMessage netty/to-byte-buf)))
-    netty/wrap-future
-    (fn [_] (netty/close ctx))))
+                      (invalid-value-response req rsp error-logger))
+                    error-logger))))))))))
+
+(defn handle-exception [error-logger ctx ex]
+  (try
+    (error-logger ex)
+    (catch Throwable e
+      (log/warn "exception in error logger" e)))
```

Comment:
Do you ignore the stacktrace for this exception on purpose? What I mean is that `(log/warn e "error in error logger")` would print both the exception message and its trace, but your code will only print the message itself.